source/fuzz/fuzzer_util.cpp

```diff
@@ -1002,6 +1024,31 @@ uint32_t MaybeGetIntegerConstant(
   return 0;
 }
 
+uint32_t MaybeGetIntegerConstantFromValueAndType(opt::IRContext* ir_context,
+                                                 uint32_t value,
+                                                 uint32_t int_type_id) {
+  auto int_type_inst = ir_context->get_def_use_mgr()->GetDef(int_type_id);
+
+  assert(int_type_inst && "The given type id must exist.");
+
+  auto int_type = ir_context->get_type_mgr()
+                      ->GetType(int_type_inst->result_id())
+                      ->AsInteger();
+
+  assert(int_type && "The given type id must correspond to an integer type.");
```

Comment:
If you do change the method contract to require that the integer is 32-bit, it could be asserted here too.

Reply:
Done